scripts/ci-report/fetch_stats_from_ci.sh

```diff
@@ -0,0 +1,139 @@
+		continue
+	fi
+
+	run_jobs_file=run-"${database_id}"-"${event}"-jobs.json
+	if [[ ! -f "${run_jobs_file}" ]]; then
+		echo "Fetching jobs for run: ${display_title} (${database_id}, ${event}, ${head_branch})"
+		gh run view "${database_id}" --json jobs >"${run_jobs_file}" || {
+			rm -f "${run_jobs_file}"
+			exit 1
+		}
+	fi
+
+	jobs="$(
+		jq -r '.jobs[] | select(.name | startswith("test-go")) | select(.status == "completed") | select(.conclusion == "success" or .conclusion == "failure") | [.databaseId, .startedAt, .completedAt, .name, .url] | @tsv' \
+			<"${run_jobs_file}"
+	)"
+
+	while read -r job; do
+		mapfile -d $'\t' -t parts <<<"${job}"
+		parts[-1]="${parts[-1]%$'\n'}"
+
+		job_database_id="${parts[0]}"
+		job_started_at="${parts[1]}"
+		job_completed_at="${parts[2]}"
+		job_name="${parts[3]}"
+		job_url="${parts[4]}"
+
+		job_log=run-"${database_id}"-job-"${job_database_id}"-"${job_name}".log
+		if [[ ! -f "${job_log}" ]]; then
+			echo "Fetching log for: ${job_name} (${job_database_id}, ${job_url})"
+			# Example log (partial).
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:18.4063489Z ##[group]Run # Artifacts are not available after rerunning a job,
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:18.4063872Z # Artifacts are not available after rerunning a job,
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:18.4064188Z # so we need to print the test stats to the log.
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:18.4064642Z go run ./scripts/ci-report/main.go gotests.json | tee gotests_stats.json
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:18.4110112Z shell: /usr/bin/bash -e {0}
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:18.4110364Z ##[endgroup]
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:19.3440469Z {
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:19.3441078Z   "packages": [
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:19.3441448Z     {
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:19.3442927Z       "name": "agent",
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:19.3443311Z       "time": 17.538
+			#   test-go (ubuntu-latest)	Print test stats	2023-04-11T03:02:19.3444048Z     },
+			#   ...
+			gh run view --job "${job_database_id}" --log >"${job_log}" || {
+				# Sometimes gh fails to extract ZIP, etc. :'(
+				rm -f "${job_log}"
+				echo "Failed to fetch log for: ${job_name} (${job_database_id}, ${job_url}), skipping..."
+				continue
+			}
+			log_lines="$(wc -l "${job_log}" | awk '{print $1}')"
+			if [[ ${log_lines} -lt 2 ]]; then
+				# Sometimes gh returns nothing and gives no error :'(
+				rm -f "${job_log}"
+				echo "Log is empty for: ${job_name} (${job_database_id}, ${job_url}), skipping..."
+				continue
+			fi
+		fi
+
+		if ! job_stats="$(
+			grep "${job_name}.*${job_step_name}" "${job_log}" |
```

Comment:
nit: Could you leave in a comment any sample line this is processing?

Reply:
The job log output format is in a comment above (by the `gh` command).